src/main/java/org/springframework/social/partnercenter/api/uri/UriProvider.java

```diff
@@ -68,6 +79,39 @@ public String buildPartnerCenterTokenUri() {
 				.build().toString();
 	}
 
+	public String buildAuthorizeUrl(String clientId, String redirectUri, String state) {
+		final OAuth2Parameters parameters = new OAuth2Parameters(Stream.of(
+				new AbstractMap.SimpleEntry<>("redirect_uri", Lists.newArrayList(redirectUri)),
+				new AbstractMap.SimpleEntry<>("state", Lists.newArrayList(state)))
+				.collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue)));
+		return buildAuthorizeUrl(clientId, parameters);
+	}
+
+	public String buildAuthorizeUrl(String clientId, OAuth2Parameters parameters) {
+		final UriComponentsBuilder uriComponentsBuilder = UriComponentsBuilder.fromUriString("https://login.microsoftonline.com").pathSegment("common", "oauth2", "authorize");
+
+		final String clientSecretKey = "client_secret";
+		if (parameters.containsKey(clientSecretKey)) {
+			uriComponentsBuilder.queryParam(clientSecretKey, parameters.getFirst(clientSecretKey));
+		}
+
+		uriComponentsBuilder.queryParam("response_mode", "form_post")
+				.queryParam("response_type", "code id_token")
```

Comment:
It would be nice to allow users to send the response_type in their parameters.

Reply:
Maybe add an issue for that in the future

Reply:
The response_type can be overwritten as part of the oauth parameters if they use the more flexible version of the function.